test_conformance/conversions/test_conversions.cpp

```diff
@@ -760,977 +510,4 @@ test_status InitCL(cl_device_id device)
-            }
-        }
-    }
-
-    if (memcmp(mapped, gRef, count * gTypeSizes[outType]))
-        info->result = gCheckResults[outType](mapped, gRef, gAllowZ, count,
-                                              vectorSizes[vectorSize]);
-    else
-        info->result = 0;
-
-    // Fill the output buffer with junk and release it
-    {
-        cl_uint pattern = 0xffffdead;
-        memset_pattern4(mapped, &pattern, count * gTypeSizes[outType]);
-        if ((error = clEnqueueUnmapMemObject(gQueue, gOutBuffers[vectorSize],
-                                             mapped, 0, NULL, NULL)))
-        {
-            vlog_error("ERROR: clEnqueueUnmapMemObject failed in "
-                       "CalcReferenceValuesComplete  (%d)\n",
-                       error);
-            gFailCount++;
-        }
-    }
-
-    if (1 == ThreadPool_AtomicAdd(&info->parent->barrierCount, -1))
-    {
-        if ((status = clSetUserEventStatus(doneBarrier, CL_COMPLETE)))
-        {
-            vlog_error("ERROR: clSetUserEventStatus failed in "
-                       "CalcReferenceValuesComplete (err: %d). We're probably "
-                       "going to deadlock.\n",
-                       status);
-            gFailCount++;
-            return;
-        }
-
-        if ((status = clReleaseEvent(doneBarrier)))
-        {
-            vlog_error("ERROR: clReleaseEvent failed in "
-                       "CalcReferenceValuesComplete (err: %d).\n",
-                       status);
-            gFailCount++;
-            return;
-        }
-    }
-    // e was already released by WriteInputBufferComplete. It should be
-    // destroyed automatically soon after all the calls to
-    // CalcReferenceValuesComplete exit.
-}
-
-static cl_program MakeProgram(Type outType, Type inType, SaturationMode sat,
-                              RoundingMode round, int vectorSize,
-                              cl_kernel *outKernel)
-{
-    cl_program program;
-    char testName[256];
-    int error = 0;
-
-    std::ostringstream source;
-    if (outType == kdouble || inType == kdouble)
-        source << "#pragma OPENCL EXTENSION cl_khr_fp64 : enable\n";
-
-    // Create the program. This is a bit complicated because we are trying to
-    // avoid byte and short stores.
-    if (0 == vectorSize)
-    {
-        // Create the type names.
-        char inName[32];
-        char outName[32];
-        strncpy(inName, gTypeNames[inType], sizeof(inName));
-        strncpy(outName, gTypeNames[outType], sizeof(outName));
-        sprintf(testName, "test_implicit_%s_%s", outName, inName);
-
-        source << "__kernel void " << testName << "( __global " << inName
-               << " *src, __global " << outName << " *dest )\n";
-        source << "{\n";
-        source << "   size_t i = get_global_id(0);\n";
-        source << "   dest[i] =  src[i];\n";
-        source << "}\n";
-
-        vlog("Building implicit %s -> %s conversion test\n", gTypeNames[inType],
```

Comment:
The new version seems to be doing fewer of these implicit conversion tests.

Reply:
From original PR:
![image](https://github.com/KhronosGroup/OpenCL-CTS/assets/11174520/9f53e348-fb39-4982-a75e-41c930570e74)
I will complement description in this PR as well

Reply:
I've restored original capacity

Reply:
Thanks; but with your latest patch, I am still seeing only one implicit conversion being tested (the `uchar -> uchar` case).  All other implicit conversion tests seem to be missing, such as `char -> uchar`, `uchar -> ulong`, etc.

The quickest way to see this is to run the test in link check mode (with the `-l` option) and compare the log with a reference log.

Reply:
@svenvh  good find, I've corrected and compared results which now looks the same, thanks.

Reply:
Looks better now, thanks.